src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
+        public void Test_with_zero_test_cases_should_not_be_run()
```

Comment:
I might be out of line here, but the rest of this test class does not use underscores between words, so I think if/when this gets accepted you might want to consider renaming the methods to match the convention in the rest of the test class.